### PR TITLE
Fix issue with PID entries in Redis not being deleted for failed jobs

### PR DESCRIPTION
**Pull request by martinbutt**

### CODE REVIEW

File: lib/Resque/Job.php [hunk 1]
```diff
@@ -242,6 +242,11 @@ public function fail($exception)
 				$this->queue
 			);
 		}
+		
+		if(!empty($job->payload['id'])) {
```
Comment (Hikariii) [inline lib/Resque/Job.php:246]: There is a lot to be said about this but primarily there is no reference to $job at this point.
So it should at least be $this.
My other comment would be that cleaning up redis Id's happens in the worker and failing a job also happens in the worker. So to me it makes more sense to cleanup this id in the worker.

Comment (martinbutt) [reply lib/Resque/Job.php]: I've fixed the variable name.

I could move this code, and the stat incrementations to a new location that listens for the onFailure event?

**Review by danhunsaker — DISMISSED** [review]

**Comment by danhunsaker** [comment]
@martinbutt Can you rebase your change against develop instead of main? I'd like to merge this, but can't cleanly do so.

**Comment by martinbutt** [comment]
@danhunsaker new PR here: https://github.com/resque/php-resque/pull/73

**Pull request closed without merging**